tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,352 @@
+    }
+
+}
+
+sub destroy_vir_network {
+
+    my $wait_script       = "30";
+    my $get_vnet_name     = "virsh net-list --all| tail -2| head -1| awk \'{print \$1}\'";
+    my $vnet_name         = script_output($get_vnet_name, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my @vnet_name_array = split(/\n+/, $vnet_name);
+    foreach (@vnet_name_array) {script_run("virsh net-destroy $_"); script_run("rm -rf $_.xml");}
+
+    my $default_path  = "/root/libvirt_default.xml";
+    if (script_run("[[ -f $default_path ]]") == 0){assert_script_run("virsh net-define $default_path",60);assert_script_run("rm -rf $default_path");}
+
+}
+
+sub run {
+
+    #Prepare VM HOST SERVER Network Interface Configuration
+    #for libvirt virtual network testing 
+    prepare_network;
+
+    #Debug br123
+    restore_standalone;
+
+    #Remove SSH old files and Generate the key pair
+    assert_script_run "rm -rf ~/.ssh/* || true;ssh-keygen -t rsa -P '' -f ~/.ssh/id_rsa";
+ 
+    #Install required packages
+    zypper_call '-t in iptables iputils bind-utils sshpass';
+
+    #Guest SSH Setup
+    my $gi_guest = '';
+    foreach my $guest (keys %xen::guests) {
+        #Archive deployed Guests
+        assert_script_run("virsh dumpxml $guest > $guest.xml");
+        record_info "$guest", "Establishing SSH connection to $guest";
+        assert_script_run("virsh start $guest", 60);
+        save_screenshot;
+        $gi_guest = script_output("sleep 40;journalctl | grep dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+	#Copy the VM host SSH Key to guest systems
+        assert_script_run("sshpass -p novell ssh-copy-id -f root\@$gi_guest", 60);
+	#Prepare the new guest network interface files for libvirt virtual netwok
+        assert_script_run("ssh root\@$gi_guest 'cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-eth1;cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-eth2'");
+
+     }
+
+##BRIDGE - USING AN EXISTING BRIDGE ON VM HOST SERVER
+##Prepare Test
+my $vnet_host_bridge = <<'EOF';
+<network>
+<name>vnet_host_bridge</name>
+<forward mode="bridge"/>
+<bridge name="BRI"/>
+</network>
+EOF
+
+    #Create HOST BRIDGE NETWORK
+    assert_script_run("echo '$vnet_host_bridge' >> vnet_host_bridge.xml");
+    assert_script_run("sed -i -e 's/BRI/$virt_host_bridge/' vnet_host_bridge.xml");
+    assert_script_run("virsh net-create vnet_host_bridge.xml");
+    assert_script_run("virsh net-list --all|grep vnet_host_bridge");
+    save_screenshot;
+    upload_logs "vnet_host_bridge.xml";
+    assert_script_run("rm -rf vnet_host_bridge.xml");
+
+    my $gi_host_bridge = '';
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "HOST BRIDGE NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_host_bridge --live");
+	assert_script_run("ssh root\@$gi_guest 'rcnetwork restart'", 60);
+        if (check_var('XEN', '1')) {
```

Comment:
I'd like suggest you'd better not use 'XEN' settings in the test any more. I only saw this setting is not often used in our test script. We usually use "SYSTEM_ROLE" instead. or using "if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen"))" is safer.

Reply:
Got it. Will update it as your mentioned.

Reply:
DONE